components/cam/src/dynamics/se/restart_dynamics.F90

```diff
@@ -359,13 +360,13 @@ function get_restart_decomp(elem, lev) result(ldof)
     do k=1,lev
        do ie=1,nelemd
           do i=1,np*np
-             ldof(j) = (elem(ie)%GlobalID-1)*np*np+(k-1)*nelem*np*np+i
+             ldof(j) = int(elem(ie)%GlobalID-1, pio_offset_kind)*np*np &
```

Comment:
Why is conversion done only on 1st multipliers, not, say, on np or nelem or i?

Reply:
Converting any of them will result in the rest being promoted.

Reply:
Some compilers like IBM XL are more picky. So, if you wish data type conversion on all, it's better to be explicit.

Reply:
we do this throughout out the code, and it's a long established part of the fortran standard.  IBM XL should be ok with it.